### PR TITLE
Flush output files in classify.

### PR DESCRIPTION
**Pull request by blankenberg**
Resolves issue where output files are erroneously empty.

sample test input:
```
>1
TAACCCTGAGCGTGCTTTCGATACGGGTTGACTTGAGGAAGGTAGGGGAGAATGGAATTCCTGGTGGAGCGGTGGAATGCGCAGATATCAGGAGGACACCAGTGGCGAAGGCGGTTCTCTGGGCCTTTCCTGACGCTGAGGAGCGAAAGCGTGGGGAGCGAACAGGCTTAGATACCCTGGTAGTCCACGCTGTAAACGGTGGGTACTAGGTGTGGGGTCCATTCCACGGGTTCCGTGCCGTAGCTAACGCTTTAAGTACCCCGCCTGGGGAGTACGGCCGCAAGGCTAAA
>2
TGGGGAGGTACGGCCGCAAGGCTAAA
>3
CGCTGGCGGCGTGCTTAACACATGCAAGTCGACGGAAAGGCCTGCTTTGTGGGTGCTCGAGTGGCGAACGGGTGAGTAACACGTGAGTAACCTGCCCTTGACTTTGGGATAACTTCAGGAAACTGGGGCTAATACCGGATAGGAGCTCCTGCTGCATGGTGGGGGTTGGAAAGTTTCGGCGGTTGGGGATGGACTCGCGGCTTATCAGCTTGTTGGTGGGGTAGTGGCTTACCAAGGCTTTGACGGGTAGCCGGCCTGAGAGGGTGACCGGCCACATTGGGACTGAGATACGGCCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGGAAGCCTGATGCAGCAACGCCGCGTGCGGGATGACGGCCTTCGGGTTGTAAACCGCTTTCGCCTGTGACGAAGCGTGAGTGACGGTAATGGGTAAAGAAGCACCGGCTAACTACGTG
```

### CODE REVIEW

**Comment by blankenberg** [comment]
@DerrickWood Is any feedback or review possible?

This resolves a critically important issue that we’ve encountered where Kraken is not outputting all of the reads under certain conditions. 

We noticed that with smaller numbers of input reads, we were ending up with empty result files (despite e.g. the kraken stderr saying 2 reads are classified, when using `--output`). This was experienced on several different systems.

I was able to resolve this by manually flushing the output files. 

Additionally, It is possible that a handful of reads are also being dropped, under certain conditions, even when some outputs are being recorded — however, I did not look or test for this case.

A workaround is to use redirection (`>`) for the classification, however, the files for ` --classified-out ` and `--unclassified-out` cannot be worked-around in this way.

An example: https://test.galaxyproject.org/u/dan/h/kraken-issues

**tldr: You cannot trust the output of Kraken when using `--output` or ` --classified-out ` or `--unclassified-out`.**

**Comment by DerrickWood** [comment]
Sorry for the delay in responding here - this looks like a reasonable fix. Thanks!

**Comment by blankenberg** [comment]
Awesome, thanks!
